wicket-extensions/src/main/java/org/apache/wicket/extensions/ajax/markup/html/AjaxLazyLoadPanel.java

```diff
@@ -139,56 +182,176 @@ protected void handleCallbackScript(final IHeaderResponse response,
 	@Override
 	protected void onBeforeRender()
 	{
-		if (state == 0)
+		if (state == LoadingState.WAITING_FOR_LAZY_COMPONENT)
 		{
 			add(getLoadingComponent(LAZY_LOAD_COMPONENT_ID));
-			setState((byte)1);
+			setState(LoadingState.REPLACING_LAZY_COMPONENT);
 		}
 		super.onBeforeRender();
 	}
 
+	@Override
+	public void onEvent(IEvent<?> event)
+	{
+		super.onEvent(event);
+
+		if (state == LoadingState.REPLACING_LAZY_COMPONENT)
```

Comment:
Shouldn't #onEvent() check for a specific event? If other events are sent through the component hierarchy (e.g. when any other Ajax request is processed), the lazy component might be replaced prematurely.